src/Core/Grid/Data/Factory/CartGridDataFactory.php

```diff
@@ -0,0 +1,171 @@
+use PrestaShop\PrestaShop\Core\Domain\Cart\CartStatus;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartForViewing;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Localization\LocaleInterface;
+use PrestaShopBundle\Translation\TranslatorInterface;
+
+/**
+ * Gets data for cart grid
+ */
+class CartGridDataFactory implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $cartDataFactory;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var LocaleInterface
+     */
+    private $locale;
+
+    /**
+     * @var string
+     */
+    private $contextCurrencyIsoCode;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @param GridDataFactoryInterface $cartDataFactory
+     * @param TranslatorInterface $translator
+     * @param LocaleInterface $locale
+     * @param $contextCurrencyIsoCode
+     * @param CommandBusInterface $queryBus
+     */
+    public function __construct(
+        GridDataFactoryInterface $cartDataFactory,
+        TranslatorInterface $translator,
+        LocaleInterface $locale,
+        $contextCurrencyIsoCode,
+        CommandBusInterface $queryBus
+    ) {
+        $this->cartDataFactory = $cartDataFactory;
+        $this->translator = $translator;
+        $this->locale = $locale;
+        $this->contextCurrencyIsoCode = $contextCurrencyIsoCode;
+        $this->queryBus = $queryBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $cartData = $this->cartDataFactory->getData($searchCriteria);
+        $modifiedRecords = $this->applyModifications($cartData->getRecords());
+
+        return new GridData(
+            $modifiedRecords,
+            $cartData->getRecordsTotal(),
+            $cartData->getQuery()
+        );
+    }
+
+    /**
+     * Modify all records in grid data.
+     *
+     * @param RecordCollectionInterface $records
+     *
+     * @return RecordCollectionInterface
+     */
+    private function applyModifications(RecordCollectionInterface $records): RecordCollectionInterface
+    {
+        $modifiedRecords = [];
+        foreach ($records as $record) {
+            $modifiedRecords[] = $this->setRecordData($record);
+        }
+
+        return new RecordCollection($modifiedRecords);
+    }
+
+    /**
+     * Modify cart record for grid data.
+     *
+     * @param array<string, mixed> $record
+     *
+     * @return array<string, mixed>
+     */
+    private function setRecordData(array $record): array
+    {
+        $cartForViewing = $this->queryBus->handle(new GetCartForViewing((int) $record['id_cart']));
```

Comment:
That said after looking into it I see that it returns the cart toal that depends on all the product's prices, and I guess also the different cart rules applied on it. I can see why it's not possible to get it in a simple query builder 

Although it could be done with less data than the handler but it would rely on legacy code so not ideal either, I guess when the price computing feature will have been refactored this part can be improved, but for a first implementation this should be enough